### PR TITLE
2d/platformer works flawlessly on Godot 3.1 beta 8

### PR DESCRIPTION
**Pull request by TestDotCom**
* fix warnings for inferred node type
* every script has a class_name
* formatted code for better clarity
* each files has its own subfolder

### CODE REVIEW

File: 2d/platformer/enemy/enemy.gd [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 var direction = -1
-var anim=""
-
+var anim = ""
+
+# state machine
+enum { 
+	STATE_WALKING = 0, 
+	STATE_KILLED = 1, 
+	WALK_SPEED = 70
```
Comment (akien-mga) [inline 2d/platformer/enemy/enemy.gd]: Walk speed is a state, it should stay a constant.

File: 2d/platformer/enemy/enemy.gd [hunk 2]
```diff
@@ -1,17 +1,21 @@
+class_name Enemy
```
Comment (akien-mga) [inline 2d/platformer/enemy/enemy.gd]: `class_name` should come after `extends`.

Comment (akien-mga) [reply 2d/platformer/enemy/enemy.gd]: At least I think :D

File: 2d/platformer/coin/coin.tscn [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
-[ext_resource path="res://coin.gd" type="Script" id=1]
-[ext_resource path="res://coin.png" type="Texture" id=2]
-[ext_resource path="res://sound_coin.wav" type="AudioStream" id=3]
-[ext_resource path="res://bullet.png" type="Texture" id=4]
+[ext_resource path="res://coin/coin.gd" type="Script" id=1]
+[ext_resource path="res://coin/coin.png" type="Texture" id=2]
+[ext_resource path="res://audio/sound_coin.wav" type="AudioStream" id=3]
+[ext_resource path="res://player/bullet.png" type="Texture" id=4]
+
```
Comment (akien-mga) [inline 2d/platformer/coin/coin.tscn]: Those added lines are weird, did the engine add this?

File: 2d/platformer/player/player.gd [hunk 4]
```diff
@@ -10,13 +13,15 @@ const BULLET_VELOCITY = 1000
...
 var linear_vel = Vector2()
-var on_floor = false
 var shoot_time = 99999 # time since last shot
 
 var anim = ""
 
 # cache the sprite here for fast access (we will set scale to flip it often)
 onready var sprite = $sprite
+# cache bullet for fast access
+onready var bullet = preload("res://player/bullet.tscn")
```
Comment (akien-mga) [inline 2d/platformer/player/player.gd]: I don't think it has to be `onready`, a `var` should be sufficient. Style wise, such a preloaded PackedScene can also be considered as a class, and should thus use PascalCase naming, so `Bullet` (or `BulletScene`).

File: 2d/platformer/player/player.gd [hunk 5]
```diff
@@ -46,16 +51,16 @@ func _physics_process(delta):
 	# Jumping
 	if on_floor and Input.is_action_just_pressed("jump"):
 		linear_vel.y = -JUMP_SPEED
-		$sound_jump.play()
+		($sound_jump as AudioStreamPlayer2D).play()
```
Comment (akien-mga) [inline 2d/platformer/player/player.gd]: Are these casts really necessary?

File: 2d/platformer/player/player.gd [hunk 6]
```diff
@@ -46,16 +51,16 @@ func _physics_process(delta):
...
 
 	# Shooting
 	if Input.is_action_just_pressed("shoot"):
-		var bullet = preload("res://bullet.tscn").instance()
-		bullet.position = $sprite/bullet_shoot.global_position # use node for shoot position
-		bullet.linear_velocity = Vector2(sprite.scale.x * BULLET_VELOCITY, 0)
-		bullet.add_collision_exception_with(self) # don't want player to collide with bullet
-		get_parent().add_child(bullet) # don't want bullet to move with me, so add it as child of parent
-		$sound_shoot.play()
+		var newBullet = bullet.instance()
```
Comment (akien-mga) [inline 2d/platformer/player/player.gd]: We use `snake_case` for identifiers, so it should be `new_bullet` (or `bullet` after having renamed the preloaded PackedScene to `Bullet` or `BulletScene`).

File: 2d/platformer/enemy/enemy.gd (hunk 1 above)
Comment (TestDotCom) [reply 2d/platformer/enemy/enemy.gd]: Got it

File: 2d/platformer/coin/coin.tscn (hunk 3 above)
Comment (TestDotCom) [reply 2d/platformer/coin/coin.tscn]: I haven't added them, so it was the engine

File: 2d/platformer/player/player.gd (hunk 4 above)
Comment (TestDotCom) [reply 2d/platformer/player/player.gd]: Got it

**Comment by akien-mga** [comment]
Great work! I agree with most changes, see my comments on scripts for a few pending issues.

Once this is finalized and merged, would you mind doing the same changes to `2d/physics_platformer`?

CC @NathanLovato @cbscribe for further style/best practice review.

File: 2d/platformer/player/player.gd (hunk 5 above)
Comment (TestDotCom) [reply 2d/platformer/player/player.gd]: Yes, otherwise the engine complains about a Node2d inferred type

**Comment by TestDotCom** [comment]
@akien-mga now I'm going to push your suggestion;
and yes I'll gladly review that one

**Comment by cbscribe** [comment]
Naming conventions call for node names to use PascalCase: "anim" -> "Anim", "camera" -> "Camera", etc.

**Comment by akien-mga** [comment]
Thanks a lot!

**Pull request merged**